### ISSUE TITLE
Consider adding a flag for user to configure normalization with the `embed` command

### ISSUE THREAD

**Issue by thompsonmj** [Imageomics/pybioclip#164]
Embeddings currently come out at [raw magnitude](https://github.com/Imageomics/pybioclip/blob/5f13513e00d9740deddc157ebb0172e787568dbb/src/bioclip/__main__.py#L102).

This is probably a reasonable default, preserving any information in the norm in the (maybe unexpected) case that the norm isn't purely noise ...

Although in training and inference, L2 norm is applied for cosine similarity. Having `pybioclip` optionally give normalized embeddings would most accurately present outputs in the geometry BioCLIP intends.

While normalization is simple post hoc, it might not be an obvious step.

Proposing the option of a flag for something like the following to configure [`normalize=True/False`](https://github.com/Imageomics/pybioclip/blob/5f13513e00d9740deddc157ebb0172e787568dbb/src/bioclip/__main__.py#L102):
```
bioclip embed --normalize=True Ursus-arctos.jpeg # sets normalize=True
bioclip embed --normalize=False Ursus-arctos.jpeg # sets normalize=False
bioclip embed Ursus-arctos.jpeg # default behavior, sets normalize=False
```
(or perhaps a simple boolean `--normalize` where present is `True`, absent is `False`)

Along with a brief CLI description and perhaps an exposition in docs.

Strongest case for keeping the default behavior of `normalize=False` I think is to avoid a breaking change.

This was motivated by an error `Floating point exception (core dumped)` during UMAP on raw embeddings, which might be due to numerically instability with large magnitude embeddings.

**Comment by hlapp** [comment]
@netzissou maybe you can comment too from the perspective of what the embedding explorer would need or want? (Also related therefore to #177.)

**Comment by NetZissou** [comment]
@hlapp the projection and clustering in `emb-explorer` always operate on L2-normalized vectors regardless of the backend (sklearn, FAISS, cuML). 

This was also motivated from a previous debugging effort. `cuML` was giving SIGFPE error from processing large magnitudes, and applying L2  norm projecting vectors onto the unit hypersphere solved the problem. We've also checked with Jianyang to ensure that L2-normalized vectors would preserve the model's native distance metric.

**Comment by hlapp** [comment]
So it sounds like `--normalize` (meaning a boolean) is all we need; if provided, the embeddings would be output in L2 norm.

@thompsonmj you haven't responded yet re: my question above, but if you don't fully agree with this conclusion, please comment soon.

**Comment by thompsonmj** [comment]
Apologies for letting this slip.

After thinking on this a bit, it appears that raw embeddings can be useful for downstream tasks like probing. Also, I think adding normalization to the `embed` command was framed as a false dichotomous choice. In order to get both the model-objective-function-native output of an L2-normalized embedding plus a raw embedding, we could simply have the `embed` command emit an L2-normalized embedding and an additional scalar for the norm, then the raw embedding could be recoverable.

I think perhaps the cleanest implementation from where the package stands now might be:

Rather than just a `--normalize` presence-only Boolean, I would change my original proposal to instead opt for a `--normalize`/`--no-normalize` toggle with one of them serving as the default behavior when neither option was passed using `argparse.BooleanOptionalAction`. If we switch default behavior in the future, anyone who explicitly adopts `--no-normalize` has stable behavior.

So for now, the default behavior can probably stay as `--no-normalize` to avoid surprises. But it can issue a notice when neither flag is specified saying something like:
> NOTE: `bioclip embed` currently returns raw (un-normalized) embeddings by default. In a future version, the default
  will change to L2-normalized embeddings (which match BioCLIP's training objective) along with the L2 norm. To opt in now, pass `--normalize`. To preserve the current behavior, pass `--no-normalize`. This warning will disappear once you pass either flag explicitly.

So `--normalize` corresponds with `True`, `--no-normalize` with `False`, and `None` if neither is passed. 

In a future major release, the default can flip to `--normalize` to align with the loss function the model was trained against.

On any alt normalizations, I can't think of a good reason now to support something besides L2 for the open clip family of models. Other norms could be computable, but I don't think they match the geometry the models are trained in.

**Comment by hlapp** [comment]
Adding the flag could be a patch-release compatible change, but adding the warning would need a point release (such as going from 2.1.x to 2.2.x).

So if we want to keep adding the flag and the warning together, then I think this needs to wait until we're rolling a 2.2.0 release. (Which we may well be soon.)
